src/components/Editor/QueryEditor.jsx

```diff
@@ -0,0 +1,50 @@
+import {required, SimpleForm, TextInput} from "react-admin";
+import config from "../../config";
+import {useEffect, useState} from "react";
+
+function QueryEditor(props) {
+
+    let {newQuery} = props
+
+    const [queryText, setQueryText] = useState()
+
+    let query;
+    if (!newQuery) {
+        query = config.queries.filter(
+            (query) => query.id === props.id
+        )[0];
+    }
+
+    useEffect(() => {
+        if (query) {
+            fetch(`${config.queryFolder}${query.queryLocation}`)
+                .then((response) =>
+                response.text())
+                .then((text) =>
+                setQueryText(text));
+        }
+    }, [])
+
+
+    const onSubmit = (data) => {
+        if (newQuery) {
+            // Create new query
+        } else {
+            // Edit existing query
+        }
+    }
+
+    return (
+        <SimpleForm onSubmit={onSubmit}>
+            <TextInput source="id" name={"Query ID"} validate={[required()]} defaultValue={query ? query.id : undefined}/>
+            <TextInput source="name" name={"Name"} validate={[required()]} defaultValue={query ? query.name : undefined}/>
+            <TextInput source="description" name={"Description"} defaultValue={query ? query.description : undefined}/>
+            <TextInput source="query" multiline={true} name={"Query"} defaultValue={queryText}/>
+            ${queryText}
+            <TextInput source="truetext" multiline={true} name={"True text"} defaultValue={query ? (query.askQuery ? query.trueText : undefined) : undefined}/>
+            <TextInput source="falsetext" multiline={true} name={"False text"} defaultValue={query ? (query.askQuery ? query.falseText : undefined) : undefined}/>
```

Comment:
All input fields should be properly styled/themed to be longer and not have the label/name overlap with the input content itself

Reply:
At least the label/name overlap may be solved after merging the main branch. I saw that in the templated queries form too and solved it there. It was the login form's css that destroyed it as a side effect by the way.